redash/models.py

```diff
@@ -830,6 +830,10 @@ def recent(cls, group_ids, user_id=None, limit=20):
 
         return query
 
+    @classmethod
+    def get_by_id(cls, _id):
+        return db.session.query(cls).filter(cls.id == _id).one()
```

Comment:
The `[]` accessor, turns `cls.query` into a `SELECT *` query...

Your original approach is the way to go then. Just if you don't mind change it to `cls.query` instead of `db.session.query`.